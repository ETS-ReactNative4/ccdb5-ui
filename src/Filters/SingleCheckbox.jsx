import { changeFlagFilter } from '../actions/filter'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import React from 'react'
import { shortIsoFormat } from './utils'

export class SingleCheckbox extends React.Component {
  constructor( props ) {
    super( props )

    this.state = { is_checked: this.props.is_checked }
  }

  componentWillReceiveProps( nextProps ) {
    const newState = {
      is_checked: nextProps.is_checked
    }

    this.setState( newState )
  }

  render() {
    return (
      <section className="single-checkbox">
        <h5>{this.props.title}</h5>
        <div className="m-form-field m-form-field__checkbox">
            <input className="a-checkbox"
                   type="checkbox"
                   onChange={ this._changeFlag.bind( this, this.props.fieldName ) }
                   checked={ this.state.is_checked }
                   value={ this.props.fieldName } />
            <label className="a-label" htmlFor="theCheckbox">Yes</label>
        </div>
      </section>
    )
  }

  // --------------------------------------------------------------------------
  // Helper Methods

  _changeFlag( field, event ) {
    const newState = {
      is_checked: !this.state.is_checked
    }
    newState[field] = event.target.value
    this.setState( newState )
  }
}

// ----------------------------------------------------------------------------
// Meta

SingleCheckbox.propTypes = {
  fieldName: PropTypes.string.isRequired,
  is_checked: PropTypes.bool
}

SingleCheckbox.defaultProps = {
  is_checked: false
}

export const mapStateToProps = state => ( {
  is_checked: state.query.has_narrative ? true : false
} )

export const mapDispatchToProps = dispatch => ( {
  changeFlagFilter: is_checked => {
    dispatch( changeFlagFilter( 'has_narrative', is_checked ) )
  }
} )

export default connect( mapStateToProps, mapDispatchToProps )( SingleCheckbox )
