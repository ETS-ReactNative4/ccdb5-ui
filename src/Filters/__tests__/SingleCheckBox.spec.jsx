import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import SingleCheckbox from '../SingleCheckbox'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'

function setupSnapshot() {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    query: {},
    aggs: {}
  })

  return renderer.create(
    <Provider store={store}>
      <SingleCheckbox />
    </Provider>
  )
}

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = setupSnapshot()
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  });
});

