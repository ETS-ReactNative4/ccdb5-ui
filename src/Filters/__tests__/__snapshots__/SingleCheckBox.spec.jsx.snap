exports[`initial state renders without crashing 1`] = `
<section
  className="single-checkbox">
  <h5>
    foo
  </h5>
  <div
    className="m-form-field m-form-field__checkbox">
    <input
      className="a-checkbox"
      id="theCheckbox"
      type="checkbox" />
    <label
      className="a-label"
      for="theCheckbox">
      bar
    </label>
  </div>
</section>
`;
